src/App.tsx

```diff
@@ -1,26 +1,205 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { USER_ID, createTodo, getTodos, removeTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './TodoList';
+import { Footer } from './Footer';
+import { Errors } from './types/Errors';
+import { FilterBy } from './types/FiilterBy';
+import { ErrorNotification } from './ErrorNotification';
+import { wait } from './utils/fetchClient';
+
+type FilterTheTodos = (todos: Todo[], filterBy: FilterBy) => Todo[];
+
+const getFilteredTodos: FilterTheTodos = (todos, filterBy) => {
+  let filteredTodos = todos;
+
+  if (filterBy !== FilterBy.All) {
+    filteredTodos = filteredTodos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.Active:
+          return !todo.completed;
+        case FilterBy.Completed:
+          return todo.completed;
+        default:
+          throw new Error(Errors.Unknown);
+      }
+    });
+  }
 
-const USER_ID = 0;
+  return filteredTodos;
+};
 
 export const App: React.FC = () => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [titleText, setTitleText] = useState('');
+  const [deletingIDs, setDeletingIDs] = useState<number[]>([]);
+
+  const handleClearingError = () => setErrorMessage(null);
+  const handleChangingFilterBy = (value: FilterBy) => setFilterBy(value);
+  const handleClearingCompletedTodos = () => {
+    const idsToDelete: number[] = [];
+
+    todos.forEach(todo => {
+      if (todo.completed) {
+        idsToDelete.push(todo.id);
+      }
+    });
+
+    setDeletingIDs(idsToDelete);
+
+    Promise.all(idsToDelete.map(id => removeTodo(id)))
+      .then(() => {
+        setTodos(prevTodos => prevTodos.filter(todo => !todo.completed));
+      })
+      .catch(() => {
+        setErrorMessage(Errors.Delete);
+        wait(3000).then(() => setErrorMessage(null));
```

Comment:
Create a handler that sets the error message and then removes it in 3 seconds as you have to do it for all requests
```typescript
const handleError = (errorMessage: Errors) => {
  setErrorMessage(Errors.Delete);
  wait(3000).then(() => setErrorMessage(null));
}; 
```